src/main/kotlin/io/meshcloud/dockerosb/metrics/gauge/PagedGaugeController.kt

```diff
@@ -0,0 +1,53 @@
+package io.meshcloud.dockerosb.metrics.gauge
+
+import io.meshcloud.dockerosb.metrics.MetricsProvider
+import io.meshcloud.dockerosb.metrics.MetricsResponse
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo
+import org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.methodOn
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+import java.time.Instant
+
+
+@RestController
+class PagedGaugeController(
+    private val metricsProviders: List<MetricsProvider<GaugeMetricModel>>
+) {
+
+  @GetMapping("/metrics/gauges/{serviceDefinitionId}")
+  fun getGaugeMetricValues(
+      @PathVariable("serviceDefinitionId") serviceDefinitionId: String,
+      @RequestParam from: Instant,
+      @RequestParam to: Instant
+  ): ResponseEntity<MetricsResponse<GaugeMetricModel>> {
+    return getResponse(serviceDefinitionId, from, to, 0)
+  }
+
+  @GetMapping("/metrics/gauges/{serviceDefinitionId}/{index}")
```

Comment:
d: just to be sure: serviceInstanceId will never contain a "/"?

Reply:
you are right ! i added a REGEX in the CatalogRepository to check this. So if someone tries to use a catalog.yml with non alphanumeric characters except minus(-) char, it will throw an error.